### DISCUSSION TITLE
docs?

### DISCUSSION THREAD

**Discussion by ElhamAryanpur** [FyroxEngine/Fyrox#142] [Ideas]
any way we could have documentation for this engine? or how to contribute to it?

I have been following rg3d for a while and found it through reddit, but a bit confused on how to get started around. so a documentation would be nice, or if there isn't, how to start contributing for one, like style guides, code of conduct, ....

Thanks!

Edit: by documentation, I mean like user guide similar to Godot's. Pardon me I sometimes use those two terms interchangeably and always forget :(

**Comment by mrDIMAS** [comment]
Hi! There is not much yet, rg3d has few tutorials ([1](https://rg3d.rs/tutorials/2021/03/05/tutorial1.html), [2](https://rg3d.rs/tutorials/2021/03/09/tutorial2.html), [3](https://rg3d.rs/tutorials/2021/03/11/tutorial3.html), [4](https://rg3d.rs/tutorials/2021/07/09/rpg-tutorial1.html)) and some [API documentation](https://docs.rs/rg3d/0.20.0/rg3d/). The reason why there is not much learning materials is because the API hasn't yet stabilized and it would be a hell for one person to keep in sync everything. I'm slowly writing tutorials, which should give an overview how to use the engine, however there should be user guides for particular parts of the engine (like sound, UI, AI, editor). I hope the situation will be improved over time, especially if someone would join the development process and helped the engine :)

**Comment by mrDIMAS** [comment]
Also, to contribute you could use (a bit outdated, and incomplete 😬 ) [architecture overview](https://github.com/rg3dengine/rg3d/blob/master/ARCHITECTURE.md) or just start doing something using the engine and add missing features. Actually you're right, I should just add contrubutions guide. About code of conduct - the main rule is very simple - don't be a dick and that's enough 😄

**Reply by ElhamAryanpur** [reply]
I will for sure check there. So far I'm loving the project so far!
